thewalrus/decompositions.py

```diff
@@ -112,3 +114,64 @@ def symplectic_eigenvals(cov):
     M = int(len(cov) / 2)
     D, _ = williamson(cov)
     return np.diag(D)[:M]
+
+
+def blochmessiah(S):
+    """Returns the Bloch-Messiah decomposition of a symplectic matrix S = uff @ dff @ vff
+       where uff and vff are orthogonal symplectic matrices and dff is a diagonal matrix
+       of the form diag(d1,d2,...,dn,d1^-1, d2^-1,...,dn^-1),
+
+    Args:
+        S (array[float]): 2N x 2N real symplectic matrix
+
+    Returns:
+        tupple(array[float],  : orthogonal symplectic matrix uff
+               array[float],  : diagional matrix dff
+               array[float])  : orthogonal symplectic matrix vff
+    """
+
+    N, m = S.shape
+
+    if N != m:
+        return False
+    if N % 2 != 0:
+        return False
+    if not is_symplectic(S):
+        return False
```

Comment:
I think this should raise a `ValueError` instead of returning false so that the return signature of the function is homogeneous.